src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -81,308 +93,599 @@ class ModelMethod(BaseModelMethod):
 
-        Returns:
-            (Optional[list[str]]): The resolved and sorted `libxc_names`.
-        """
-        return sorted(
-            [
-                functional.libxc_name
-                for functional in xc_functionals
-                if functional.libxc_name is not None
-            ]
-        )
+        Common choices:
+        • erf     — error function (used in LC-ωPBE, CAM-B3LYP)
+        • erfc    — complementary error function (equivalent to erf split)
+        • Yukawa  — exponential screening, e.g. HSE-style
+        • exp     — simple exponential decay
+        • Gaussian — Gaussian screening form
+        • Slater  — Slater-type exponential
 
-    def resolve_jacobs_ladder(
-        self,
-        libxc_names: list[str],
-    ) -> str:
-        """
-        Resolves the `jacobs_ladder` from the `libxc_names`. The mapping (libxc -> NOMAD) is set in `self._jacobs_ladder_map`.
+        Needed to interpret the range-separation parameter ω correctly.
+        """,
+    )
 
-        Args:
-            libxc_names (list[str]): The list of `libxc_names`.
+    weight = Quantity(
+        # type=unit_float,
+        type=np.float64,
+        description="""
+            Fractional contribution in the density functional result.
+            Only applies when the functional is part of a larger, composed functional, that is not included in the LibXC.
+            All components' fractions should sum up to one.
+         """,
+    )
 
-        Returns:
-            (str): The resolved `jacobs_ladder`.
-        """
-        if libxc_names is None:
-            return 'unavailable'
 
-        rung_order = {x: i for i, x in enumerate(self._jacobs_ladder_map.keys())}
-        re_abbrev = re.compile(r'((HYB_)?[A-Z]{3})')
+class XCFunctional(ArchiveSection):
+    """
+    Normalized XC information for a calculation (possibly multi-component).
+    """
 
-        abbrevs = []
-        for xc_name in libxc_names:
-            try:
-                abbrev = re_abbrev.match(xc_name).group(1)
-                abbrev = abbrev.lower() if abbrev == 'HYB_MGG' else abbrev[:3].lower()
-                abbrevs.append(abbrev)
-            except AttributeError:
-                continue
+    components = SubSection(sub_section=XCComponent.m_def, repeats=True)
 
-        try:
-            highest_rung_abbrev = (
-                max(abbrevs, key=lambda x: rung_order[x]) if abbrevs else None
-            )
-        except KeyError:
-            return 'unavailable'
-        return self._jacobs_ladder_map.get(highest_rung_abbrev, 'unavailable')
+    functional_key = Quantity(
+        type=str,
+        description="""
+        Canonical functional alias representing one XC functional as a whole.
+        Used for filtering and display (e.g. 'PBE', 'PBE0', 'B3LYP', 'SCAN+rVV10').
+        Typically corresponds to the original name reported in the code,
+        """,
+    )
```

Comment:
I would simply add these quantities directly under `DFT`. Structure is nice and all, but here 2 sections suffice, given that `functional_key` maps 1-to-1 the DFT method used.